benchopt/plotting/generate_html.py

```diff
@@ -0,0 +1,296 @@
+        max_rows=15,
+        nb_total_benchs=len(results),
+        last_updated=datetime.now(),
+    )
+
+
+def render_index(benchmarks):
+    return Template(filename=str(TEMPLATE_INDEX),
+                    input_encoding="utf-8").render(
+        benchmarks=benchmarks,
+        nb_total_benchs=len(benchmarks),
+        max_rows=15,
+        last_updated=datetime.now(),
+    )
+
+
+def copy_static(benchmark=None):
+    if benchmark is not None:
+        dst = benchmark / BUILD_DIR / 'static'
+    else:
+        dst = BUILD_DIR / 'static'
+    if dst.exists():
+        shutil.rmtree(dst)
+    shutil.copytree(ROOT / 'static', dst)
+
+
+def render_all_results(results, benchmark, local=True):
+    if local:
+        template = TEMPLATE_LOCAL_RESULT
+    else:
+        template = TEMPLATE_RESULT
+    htmls = []
+    for result in results:
+        html = Template(
+            filename=str(template),
+            input_encoding="utf-8").render(
+            result=result,
+            benchmark=benchmark,
+        )
+        htmls.append(html)
+    return htmls
+
+
+def plot_benchmark_html(df, output_dir, kinds,
+                        display=True, index=False):
+    """
+    Plot convergence curves and histograms for a
+    given benchmark in an HTML file using plotly.
+
+    Parameters
+    ----------
+    df : instance of pandas.DataFrame
+        The benchmark results.
+    benchmark_instance : Benchmark
+        The benchmark instance.
+    display : bool
+        If set to True, display the curves by opening
+        the default browser.
+
+    Returns
+    -------
+    None
+    """
+
+    benchmark = output_dir
+    try:
+        print("Removing old html files")
+        shutil.rmtree(benchmark / BUILD_DIR)
+    except FileNotFoundError:
+        print("Building html files")
+
+    copy_static(benchmark)
+    print(f'Rendering benchmark: {benchmark}')
+
+    Path(benchmark / BUILD_DIR).mkdir(exist_ok=True)
+    Path(benchmark / BUILD_DIR_FIGURES).mkdir(exist_ok=True, parents=True)
+    Path(benchmark / BUILD_DIR_OUTPUTS).mkdir(exist_ok=True, parents=True)
+
+    results = get_result_from_df(df, kinds)
+    htmls = render_all_results(results, benchmark)
+
+    result_filename = benchmark / BUILD_DIR / results[0]['page']
+    print(f"Writing results to {result_filename}")
+    with open(result_filename, "w") as f:
+        f.write(htmls[0])
+
+    if display:
+        webbrowser.open(str(result_filename), new=2)
+
+
+@click.command()
+@click.option('--pattern', '-k', 'patterns',
+              metavar="<pattern>", multiple=True, type=str,
+              help="Include results matching <pattern>.")
+@click.option('--benchmark', '-b', 'benchmarks',
+              metavar="<pattern>", multiple=True, type=str,
+              help="benchmarks to include.")
+@click.option('--root', type=str, default=None,
+              help="root directory for results files.")
+def plot_benchmark_html_all(patterns=(), benchmarks=(), root=None):
```

Comment:
this should be a function whithout click and a corresponding click command should be put in `benchopt/cli/process_results.py`